api/app/models/tournament_player.rb

```diff
@@ -0,0 +1,8 @@
+class TournamentPlayer < ApplicationRecord
+  enum position: Hash[%w[no_preference attacker defender].map { |key| [key, key] }]
```

Comment:
I prefer use this format:
`enum position: { attacker: 'attacker', defender: 'defender' }`
and please remove no_preference type.